src/SplitCore.hs

```diff
@@ -64,7 +64,7 @@ newTypePatternPtr = TypePatternPtr <$> newUnique
 data SplitCore = SplitCore
   { _splitCoreRoot         :: SplitCorePtr
   , _splitCoreDescendants  :: Map SplitCorePtr (CoreF TypePatternPtr PatternPtr SplitCorePtr)
-  , _splitCorePatterns     :: Map PatternPtr ConstructorPattern
+  , _splitCorePatterns     :: Map PatternPtr (ConstructorPatternF PatternPtr)
   , _splitCoreTypePatterns :: Map TypePatternPtr TypePattern
```

Comment:
Hmm, this datatype seems inconsistent (but it was already inconsistent before this PR, so we don't have to fix it now). One concern I have with those `Ptr`s is that it's not clear whether we should be looking them up in the global `ExpanderState` or in this local `Map PatternPtr`. I see that for `CorePtr`, we defined a dedicated `SplitCorePtr` in order to clarify that we should look it up in this local `Map SplitCorePtr`, not in the global `ExpanderState`. I assume the same is true for the `TypePatternPtr`s and the `PatternPtr`s, but for those we did not create a special `SplitCorePatternPtr` newtype. Why not?

Reply:
I should probably clarify that when I'm saying that "it's not clear whether we should be looking them up in the global ExpanderState or in this local `Map PatternPtr`", I mean that it won't be clear at the use site, where the definition of `SplitCore` won't be right in our face. Looking at this definition it _is_ quite clear what the intent is, my concern is that we chose types which only imprecisely reflect that intent.

Reply:
This is worth thinking about.

So far, we create the `Split...` by wrapping its record constructor around the `Map` from the global state, and accept that it contains junk. This is indeed a bit confusing.